openedx/core/djangoapps/certificates/api.py

```diff
@@ -88,11 +89,19 @@ def display_date_for_certificate(course, certificate):
 
     Arguments:
         course (CourseOverview or course descriptor): The course we're getting the date for
+        certificate (GeneratedCertificate): The certificate we're getting the date for
     Returns:
         datetime.date
     """
+    try:
+        if certificate.date_override:
```

Comment:
When my tests (where `date_override` is `True`) get to this line, `certificate.date_override` doesn’t exist. It goes on to the `except` and continues to return one of the other dates.